README.md

```diff
@@ -35,19 +40,212 @@ Outline the file contents of the repository. It helps users navigate the codebas
 
 ## Prerequisites
 
-Outline the required components and tools that a user might need to have on their machine in order to run the sample. This can be anything from frameworks, SDKs, OS versions or IDE releases.
+- [Node.js](https://nodejs.org/en/download/) must be installed to run this sample.
+- [Angular-cli](https://cli.angular.io/) must be installed to run this sample.
+- [Dotnet Core SDK](https://dotnet.microsoft.com/download) must be installed to run this sample.
+- We recommend [VS Code](https://code.visualstudio.com/download) for running and debugging this cross-platform application.
 
 ## Setup
 
-Explain how to prepare the sample once the user clones or downloads the repository. The section should outline every step necessary to install dependencies and set up any settings (for example, API keys and output folders).
+Using a command line interface such as VS Code integrated terminal, locate the application directory. Then:
+
+1. Install .NET Core API dependencies:
+
+```console
+cd TodoListApplication/TodoListAPI
+dotnet restore
+```
+
+1. Install Angular SPA dependencies:
+
+```console
+cd TodoListApplication/TodoListSPA
+npm install
+```
+
+1. Trust development certificates:
+
+```console
+cd TodoListApplication/TodoListAPI
+dotnet dev-certs https --clean
+dotnet dev-certs https --trust
+```
+
+Learn more about [HTTPS in .NET Core](https://docs.microsoft.com/en-us/aspnet/core/security/enforcing-ssl?view=aspnetcore-3.1).
+
+1. Configure application parameters:
+
+See [Registration](#Registration) for details.
 
 ## Running the sample
 
-Outline step-by-step instructions to execute the sample and see its output. Include steps for executing the sample from the IDE, starting specific services in the Azure portal or anything related to the overall launch of the code.
+Using a command line interface such as VS Code integrated terminal, locate the application directory. Then:  
+
+```console
+cd TodoListApplication/TodoListAPI
+dotnet run
+```
+
+```console
+cd TodoListApplication/TodoListSPA
+npm start
+```
+
+## Debugging the sample

```

Comment:
>sample [](start = 17, length = 6)

Add this line somewhere (check the relative link). It greatly increases the chances of developers reaching out to us instead of giving up in frustration

> Did the sample not work for you as expected? Did you encounter issues trying this sample? Then please reach out to us using the [GitHub Issues](../../../../issues) page.